### PR TITLE
Build selected build-target when starting to debug

### PR DESCRIPTION
**Pull request by Maddimax**
### This changes visible behavior

The following changes are proposed:

- When launching, not only build the selected launch target, but also the selected build target if the two differ.

## The purpose of this change

For some projects, you always want to build ALL. This allows to set ALL as the build target, but still launch the selected launch target.

### CODE REVIEW

**Comment by Maddimax** [comment]
@microsoft-github-policy-service agree

**Comment by elahehrashedi** [comment]
We don't want to just add the build target to the list. A better approach would be to inform the user that the launch target doesn't match the build targets.

File: src/cmakeProject.ts [hunk 1]
```diff
@@ -2162,7 +2162,10 @@ export class CMakeProject {
 
         const buildOnLaunch = this.workspaceContext.config.buildBeforeRun;
         if (buildOnLaunch || isReconfigurationNeeded) {
-            const buildResult = await this.build([chosen.name]);
+            const defaultBuildTargets = await this.getDefaultBuildTargets();
+            const allTargetsToBuild = new Set(defaultBuildTargets);
+            allTargetsToBuild.add(chosen.name);
```
Comment (elahehrashedi) [inline src/cmakeProject.ts:2167]: Check if the build target is ALL, we shouldn't add the launch targets to ALL.

Comment (bobbrow) [reply src/cmakeProject.ts:2167]: No, you should check if it's `this.allTargetName`.  If that's the case, then you don't need to add the debug target to the list.

Comment (Maddimax) [reply src/cmakeProject.ts]: Done.

Comment (elahehrashedi) [reply src/cmakeProject.ts]: use `this.allTargetName` instead of `all`

Comment (Maddimax) [reply src/cmakeProject.ts]: Done.

**Review by elahehrashedi — APPROVED** [review]

**Pull request merged**